### PR TITLE
LibGUI: Fix weird behavior when switching lines using Ctrl+Shift+[Up,Down]

### PR DESCRIPTION
**Pull request by LucasChollet**
On this demonstration video, Ctrl+Shift+Up is fixed, but Ctrl+Shift+Down still does the old behavior

https://user-images.githubusercontent.com/26030965/168887599-be180ca0-328c-41da-b715-b020733f462a.mp4

### CODE REVIEW

File: Userland/Libraries/LibGUI/EditingEngine.cpp [hunk 1]
```diff
@@ -97,10 +111,24 @@ bool EditingEngine::on_key(KeyEvent const& event)
...
+                selection.start().set_line(final_start_line);
+                selection.end().set_line(final_end_line);
+                m_editor->set_selection(selection);
+                m_editor->did_update_selection();
+            }
+        } else {
+            if (event.shift() && m_editor->selection().start().is_valid()) {
+                m_editor->selection().set_end(m_editor->cursor());
+                m_editor->did_update_selection();
+            }
```
Comment (petelliott) [inline Userland/Libraries/LibGUI/EditingEngine.cpp:131]: it looks like you duplicated a lot of logic here. could this be a lambda that takes in the direction the line was moved?

Comment (LucasChollet) [reply Userland/Libraries/LibGUI/EditingEngine.cpp:131]: Even more code can be merged with functions that take the direction. I'm currently working on making a `MoveLineUpOrDownCommand` to take advantages of commands, and I also ended deleting up and down functions. 

I planned to make on other PR to split logical changes and code refactoring to ease the review process. But I can definitely merge them.

**Comment by circl-lastname** [comment]
What's a Maj?

**Comment by LucasChollet** [comment]
Ohh, that's the French (and abbreviated) word for Shift. It didn't cross my mind that it's not the same in English.

File: Userland/Libraries/LibGUI/Widget.h [hunk 2]
```diff
@@ -50,6 +50,15 @@ enum class VerticalDirection {
     Down
 };
 
+constexpr VerticalDirection KeyCode_to_VerticalDirection(KeyCode const& key)
```
Comment (MacDue) [inline Userland/Libraries/LibGUI/Widget.h:53]: This method seems weirdly named, I get `KeyCode` and `VerticalDirection` are types. But using pascal case like this for functions is generally not done.

**Pull request merged**